### DISCUSSION TITLE
GPT4 Code Interpreter in LibreChat with GPT4 and other versions of GPT

### DISCUSSION THREAD

**Discussion by githubdebugger** [danny-avila/LibreChat#796] [Feature Requests & Suggestions]
I was wondering if GPT4 code interpreter feature could be added to LibreChat (as a plugin or otherwise) with GPT4 and GPT3.5 etc.?

**Comment by danny-avila** [comment]
It has to be reverse engineered but this is planned. My current timeline is finishing/improving some "playground" features, which i'm making a lot of progress on and hope to wrap up soon, then file support (chat with docs/memory), and then code-interpreter reverse engineering

**Comment by githubdebugger** [comment]
Great, glad to hear Code-Interpreter will be added to LibreChat. 

Thanks!

**Comment by grexzen** [comment]
I tried really hard to do it and was going to share it next, but the langchain backend is a nightmare to work with.

**Comment by danny-avila** [comment]
> I tried really hard to do it and was going to share it next, but the langchain backend is a nightmare to work with.

@grexzen Sorry this was your experience, do you mean my implementation and use of langchain? I can definitely be a help here. Or even if you find a good way of doing it, I can put in the work to make it work

**Reply by danny-avila** [reply]
my plan as of now is to use dominic's project, "codeinterpreter-api": https://github.com/shroominic/codeinterpreter-api

Rather than reinvent the wheel, and it's the best solution I've used.

**Reply by grexzen** [reply]
No, your app is awesome, thanks for sharing. It is more trying to understand and especially debug langchain. Now that we can call functions in the API, it is easier to do it that way.

**Comment by ronith256** [comment]
You can try this simple plug-in I made : https://github.com/ronith256/Code-Interpreter-LibreChat.git. It is very much in beta, and does not sandbox the python env. 

### Demo
<img src="https://i.ibb.co/3NTz47L/image.png" width="200" height="200" />

**Reply by danny-avila** [reply]
@ronith256 Amazing! do you mind submitting a PR? if not I can get around to adding it soon

**Reply by ronith256** [reply]
I have created a PR. I'm not sure if it's proper to the guidelines. Please check and revert on what I should update on. Thanks

**Reply by grexzen** [reply]
Awesome work.
